src/config-import.js

```diff
@@ -0,0 +1,231 @@
+import yauzl from 'yauzl-promise'
+import { validate, valueSchemas } from '@mapeo/schema'
+import { text, buffer } from 'node:stream/consumers'
+
+// Throw error if a zipfile contains more than 10,000 entries
+const MAX_ENTRIES = 10_000
+
+/**
+ * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
+ */
+/**
+ * @typedef {{
+ *   presets: { [id: string]: unknown }
+ *   fields:  { [id: string]: unknown }
+ * }} PresetsFile
+ */
+/**
+ * @typedef {Parameters<import('./icon-api.js').IconApi['create']>[0]} IconData
+ */
+
+/**
+ * @param {string} configPath
+ */
+export async function readConfig(configPath) {
+  /** @type {Error[]} */
+  const warnings = []
+
+  const zip = await yauzl.open(configPath)
+  if (zip.entryCount > MAX_ENTRIES) {
+    throw new Error(`Zip file contains too many entries. Max is ${MAX_ENTRIES}`)
+  }
+  /** @type {Entry[]} */
+  const entries = /** @type {any} */ (await zip.readEntries(MAX_ENTRIES))
+
+  const presetsEntry = entries.find((e) => e.filename === 'presets.json')
```

Comment:
I wonder if we could avoid reading all the entries into memory. Would something like this work?

```suggestion
  /** @type {undefined | Entry} */
  let presetsEntry
  for await (const entry of zip) {
    if (entry.filename === 'presents.json') {
      presetsEntry = entry
      break
    }
  }
```

See my suggestion in `icons` below.